packages/babel-parser/src/parser/expression.js

```diff
@@ -1460,6 +1460,26 @@ export default class ExpressionParser extends LValParser {
     return this.finishNode(node, "TemplateLiteral");
   }
 
+  parseKeyValuePairs() {
+    const propHash: any = Object.create(null);
+    const node = this.startNode();
+    node.properties = [];
+
+    do {
+      const prop = this.parseObjectMember();
+      // $FlowIgnore
+      if (prop.shorthand) {
+        this.addExtra(prop, "shorthand", true);
+      }
+
+      node.properties.push(prop);
+    } while (this.eat(tt.comma));
+    if (!this.match(tt.eq) && propHash.start !== undefined) {
+      this.raise(propHash.start, "Redefinition of __proto__ property");
+    }
+    return this.finishNode(node, "ObjectExpression");
```

Comment:
I don't think that module attributes allow arbitrary values (I might be wrong).

I think that this should be simplified to (pseudo-JavaScript)
```js
const attrs = [];

do {
  const node = startNode();
  parseIdentifier();
  expect(":");
  parseLiteral();
} while (eat(","));

return attrs; // Return the array of attributes, not a wrapper node.
```

It could also be inlined in `maybeParseModuleAttributes`, since it's short.

Reply:
This looks way better and simpler. I will make the changes. Thank you for the inputs 👍

Reply:
few questions here, should the above implementation look something like this
```js
const attrs = [];
do {
  const node = this.startNode();
  node.key = this.parseIdentifier();
  node.computed = false;
  node.method = false;
  node.shorthand = false;
  this.expect(tt.colon);
  node.value = this.parseLiteral(this.state.value);
  this.finishNode(node, "ObjectProperty");
  attrs.push(node);} while (this.eat(tt.comma));
```

Reply:
or should we just do
```js
node.value = this.parseLiteral(this.state.value, "Literal");
```

Reply:
The second `type` argument is meant to set type of the AST nodes. Babel supports the following `types` (but not general `"Literal"`)
```
RegExpLiteral
NullLiteral
StringLiteral
BooleanLiteral
NumericLiteral
BigIntLiteral
```
A primitive value is a member of one of the following built-in types: Undefined, Null, Boolean, Number, String, and Symbol. So you can guard against `parseExprAtom` with (pseudoJS)
```
this.match(tt._null) || ... tt._true, tt._false, tt.name (and value is "undefined"), tt.string, tt.num, tt.bigint
```

Note that `Symbol` is also a primitive type and not covered above, but I think we can forbid symbol in module attributes? @xtuc

Reply:
These changed have been updated. Thank you

Reply:
I'd recommend starting with just supporting string literals at this point.

Reply:
Made the changes and now we are only supporting String Literals.